odl/discr/lp_discr.py

```diff
@@ -89,39 +79,15 @@ def __init__(self, fspace, partition, tspace, interp='nearest', **kwargs):
             raise ValueError('`fspace.scalar_out_dtype` does not match '
                              '`tspace.dtype`: {} != {}'
                              ''.format(fspace.scalar_out_dtype, tspace.dtype))
+        if partition.shape != tspace.shape:
+            raise ValueError(
+                "`partition.shape` must be equal to `tspace.shape`, but "
+                "{} != {}".format(partition.shape, tspace.shape)
+            )
```

Comment:
Probably this was implicitly  checked in sampling and interpolation operators earlier. This version is better anyhow.